src/main/java/org/onedatashare/server/service/SSHConsoleService.java

```diff
@@ -0,0 +1,141 @@
+                sink.error(new Exception("Failed to create SSH session"));
+            }
+            session.setUserInfo(new SSHUserInfo(decryptionService.getDecryptedPassword(password)));
+            sink.success(session);
+        });
+    }    //createSession()
+
+    public Flux runCommand(UserAction ua, String commandWithPath){
+        return createSession(ua.getUri(), ua.getCredential().getUsername(), ua.getCredential().getPassword(), Integer.parseInt(ua.getPortNumber()))
+                .flux()
+                .flatMap(session -> {
+                    try {
+                        session.connect();
+                        ChannelExec channel = (ChannelExec)session.openChannel("exec");
+                        channel.setCommand(commandWithPath);
+                        return connectAndReadOutput(channel);
+                    }
+                    catch(Exception e){
+                        ODSLoggerService.logError("Error occurred while executing SSH command");
+                        e.printStackTrace();
+                        return Flux.error(new Exception("Error occurred while executing SSH command"));
+                    }
+                });
+    }
+
+    public Flux connectAndReadOutput(Channel channel){
+        try {
+            InputStream in = channel.getInputStream();
+            InputStream err = channel.getExtInputStream();
+
+            channel.connect();
+
+            StringBuilder outputBuffer = new StringBuilder();
+            StringBuilder errorBuffer = new StringBuilder();
+
+            byte[] tmp = new byte[1024];
+            while (true) {
+                while (in.available() > 0) {
+                    int i = in.read(tmp, 0, 1024);
+                    if (i < 0) break;
+                    outputBuffer.append(new String(tmp, 0, i));
+                }
+                while (err.available() > 0) {
+                    int i = err.read(tmp, 0, 1024);
+                    if (i < 0) break;
+                    errorBuffer.append(new String(tmp, 0, i));
+                }
+                if (channel.isClosed()) {
+                    if ((in.available() > 0) || (err.available() > 0))
+                        continue;
+                    break;
+                }
+                try {
+                    Thread.sleep(500);
+                } catch (Exception ee) {
+                }
+            }
+            channel.disconnect();
+            ShellCommandResponse response = new ShellCommandResponse();
+            response.setOutput(outputBuffer.toString());
+            if(errorBuffer.length() > 0){
+                response.setError(errorBuffer.toString());
+            }
+            return Flux.just(response);
+        }
+        catch (Exception e){
+            e.printStackTrace();
+            return Flux.error(new Exception("Error while Executing command"));
+        }
+    }
+
+}
+
+
+class SSHUserInfo implements UserInfo{
```

Comment:
Can we put the SSHUserInfo class into its own file? It is bad practice to have these classes laying around not in their own files unless there is a specific dependency. Java has a standard of each class in its own file.

Reply:
Done